qiskit/transpiler/passes/layout/dense_layout.py

```diff
@@ -66,11 +67,24 @@ def run(self, dag):
             raise TranspilerError(
                 "A coupling_map or target with constrained qargs is necessary to run the pass."
             )
-        layout_components = disjoint_utils.run_pass_over_connected_components(
-            dag,
-            self.coupling_map if self.target is None else self.target,
-            self._inner_run,
-        )
+        if self.target is not None:
+            layout_components = disjoint_utils.run_pass_over_connected_components(
+                dag,
+                self.target,
+                self._inner_run,
+            )
+            if layout_components is None:
+                layout_components = disjoint_py.run_pass_over_connected_components(
+                    dag,
+                    self.coupling_map,
+                    self._inner_run,
+                )
+        else:
+            layout_components = disjoint_py.run_pass_over_connected_components(
+                dag,
+                self.coupling_map,
+                self._inner_run,
+            )
```

Comment:
Is this kind of thing the only reason to keep all the Python stuff around (barring backwards compat deprecations etc)?  If so, would it be ok to get rid of the Python-space use, and pass a `Target.from_configuration(basis_gates=["cx", "u"], coupling_map=self.coupling_map)` to the Rust-space version if/when needed?  It's a bit less efficient, but that's the path we want to drive towards, I think?